src/reflection/skrypnyk/PersonFactory.java

```diff
@@ -0,0 +1,39 @@
+package src.reflection.skrypnyk;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class PersonFactory {
+
+    public static void main(String[] args) throws ClassNotFoundException, IllegalAccessException, NoSuchFieldException, InvocationTargetException, NoSuchMethodException, InstantiationException {
+        Person person = createPerson("First",1,"Lviv");
+        Person personSecond = createPerson("Second",2,"Kyiv");
+
+        printFields(person);
+        System.out.println();
+        printFields(personSecond);
+
+    }
+    public static Person createPerson(String name, int age, String address) throws InvocationTargetException, InstantiationException, IllegalAccessException {
+        try {
+            Constructor <Person> constructor = Person.class.getConstructor(String.class,int.class,String.class);
+            return constructor.newInstance(name,age,address);
+        } catch (NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+    public static void printFields(Person person){
+        Class<? extends Person> personInfo = person.getClass();
+        Field [] fieldName = personInfo.getDeclaredFields();
+        for (Field declaredField : fieldName) {
+            declaredField.setAccessible(true);
+        }
+        System.out.println("Person name: " + person.getName());
+        System.out.println("Person age: "+ person.getAge());
+        System.out.println("Person address: " + person.getAddress());
```

Comment:
Here, you use getters directly -- getters are public available functionality we can call easily.
In this homework, for practice, we have to retrieve values of fields using reflection. In for-loop, where you setting accessible, you also need after this line get the value of this field of the Person object.

```
..
declaredField.setAccessible(true);
*this you have to print field name and it's value*
using methods of Field -- declaredField.getName() (for printing what is the field) and declaredField.get(person) -- to get value
```